### ISSUE TITLE
DistributedSampler can't shuffle the dataset 

### ISSUE THREAD

**Issue by qmpzzpmq** [PyTorchLightning/pytorch-lightning#4509]
almost same things with 
https://github.com/pytorch/pytorch/issues/31771
https://github.com/huggingface/transformers/issues/3721

how to fix them in pytorch lighting

**Comment by s-rog** [comment]
PL already calls `set_epoch()` for it as part of the training loop.

I'll close this for now, if you have a bugged replicable code sample feel free to re-open.
